src/main/kotlin/org/rust/lang/core/psi/RsFile.kt

```diff
@@ -59,6 +59,7 @@ class RsFile(
         get() = superMods.lastOrNull()?.takeIf { it.isCrateRoot }
 
     override fun setName(name: String): PsiElement {
+        if (this.name == RsConstants.MOD_RS_FILE) return this
```

Comment:
How can I rename `mod.rs` now?

Reply:
Maybe this is not the best place to not rename `mod.rs` file, but anyway how I would know if user wants to rename file instead of module?

Reply:
Ok. https://github.com/intellij-rust/intellij-rust/pull/4077 has a similar issue. I will think of a solution that doesn't use a mutable state.

Reply:
@mchernyavsky I am quite sure it is not related. #4077's problem is that we don't know which reference target to choose. Here, the problem is that we don't know users intention. No matter where user invokes module renaming and what are intentions, `RsRenameProcessor` end up dealing with mod.rs file. 

As for possible solutions, I have come up to two options:
- I think if user invokes rename action from the Project View we can safely assume that he wants to rename only mod.rs file. So looks like if we create custom `RenameHandler` only for mod.rs files, there is a way to know where action was invoked from. I have tried it and `LangDataKeys.IDE_VIEW` is exactly what we need.
- We can make module references resolve to `RsModDeclItem`, while keep corresponding file as navigation target. Thus files could be renamed independently. 

If mod.rs would still be the only solution for module creation, I personally would prefer second option. But since using mod.rs is currently an unfavorable way, I think the first option is just enough.